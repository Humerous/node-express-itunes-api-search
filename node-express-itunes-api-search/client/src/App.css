body {
  background-color: #e09b04;
  margin: 2em 2em;

  display: flex;
  place-items: center;
  font-family: Arial, Helvetica, sans-serif;
}

/* FORM BELOW */

.SearchForm {
  text-align: center;
}

input[type='text'] {
  background-color: transparent;
  width: 100%;
  border: 0;
  border-bottom: solid 1px #fff;
  padding-bottom: 4px;
  color: #000;
  font-weight: lighter;
  letter-spacing: 2px;
  margin: 0 20px 30px 0;
  font-size: 20px;
}

input[type='text'] {
  outline: none;
}

input:-webkit-autofill {
  -webkit-box-shadow: 0 0 0 30px #e09b04 inset;
  -webkit-text-fill-color: #000;
}

select {
  background-color: #e09b04;
  width: 100%;
  border: 0;
  border-bottom: solid 1px #fff;
  padding-bottom: 4px;
  color: #000;
  font-weight: lighter;
  letter-spacing: 2px;
  margin: 0 20px 30px 0;
  font-size: 20px;
  cursor: pointer;
}

option {
  background-color: transparent;
  width: 50%;
  border: 0;
  padding-bottom: 4px;
  color: #e09b04;
  font-weight: lighter;
  letter-spacing: 2px;
  margin: 0 20px 30px 0;
  font-size: 20px;
}

select:focus > option:hover {
  background: #ec0101;
}

button {
  border: 0;
  border-radius: 10px;
  padding: 10px 35px;
  margin-left: 30px;
  font-weight: 700;
  letter-spacing: 1px;
  font-size: 20px;
  cursor: pointer;
}
