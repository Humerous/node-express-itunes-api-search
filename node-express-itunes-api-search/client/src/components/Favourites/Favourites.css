/* FAVOURITES SELECTS */
.FavOuterShell {
  z-index: 5;
  position: sticky;
  top: 2em;
  min-height: 100vh;
  background: #e09b04;
  color: #000;
  padding: 1em;
  margin: 0 2em 2em 1em;
}

.FavOuterContent {
  color: #ec0101;
  margin: 1em;
  padding: 1em 2em;
  border: 1px solid #fff;
}

.FavOuterHeader {
  font-size: 2em;
  font-weight: 500;
  letter-spacing: 4px;
  line-height: 2;
  color: #000;
}

.TableHead {
  color: #000 !important;
  line-height: 2;
}

.dumpsterBtn {
  cursor: pointer;
  color: #ec0101;
}
