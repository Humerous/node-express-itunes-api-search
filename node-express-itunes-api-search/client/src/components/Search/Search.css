/* SEARCH */
.SearchOuterShell {
  position: relative;
  box-sizing: border-box;
  min-height: 100vh;
  background: #e09b04;
  border-radius: 10px;
  color: #000;
  padding: 1em;
  margin: 2em 1em 2em 2em;
}

.SearchOuterContent {
  margin: 1em;
}

.SearchOuterHeader {
  background-color: #e09b04;
  z-index: 5;
  position: sticky;
  top: 0;
}

.SearchOuterHeading {
  font-size: 40px;
  font-weight: 300;
  letter-spacing: 4px;
  line-height: 2;
  color: #000;
}

.thumbBtn {
  cursor: pointer;
}
