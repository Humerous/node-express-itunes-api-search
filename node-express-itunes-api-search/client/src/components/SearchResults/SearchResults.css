/* SEARCH BAR */
.ResultOuterContainer {
  box-sizing: border-box;
  display: flex;
  justify-content: left;
  margin: 1em 0 1em 0;
}

.ResultSmallImg {
  border: 1px solid #fff;
  border-radius: 4px;
  padding: 8px;
  height: 120px;
  width: 120px;
}

.ResultSearchDetails {
  display: grid;
  box-sizing: border-box;
  width: 60%;
  line-height: 1.1;
  margin-left: 1em;
}

.ResultAllError {
  text-align: center;
  font-size: 30px;
  font-weight: 300;
  letter-spacing: 4px;
  line-height: 2;
  color: #ec0101;
}

.ResultAllDetails > p {
  margin-bottom: 0;
}

.ResultAllBtn {
  display: grid;
  place-items: top;
}

.ResultsAllTitle {
  font-weight: 600;
}

.ResultsAllImage {
  font-style: italic;
  font-weight: 400;
  color: #000;
}

.ResultsAllMedia {
  font-weight: 200;
  color: #000;
}
