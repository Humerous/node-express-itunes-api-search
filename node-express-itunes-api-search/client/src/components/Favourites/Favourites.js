import React from 'react';

// <!---- STYLE SHEET ---->
import '../Favourites/Favourites.css';

// <!---- IMPORT REACT STRAP ---->
import { Table } from 'reactstrap';

// <!---- FONT AWESOME---->
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

// <!---- SELECTED FAVOURITES FUNCTION---->
function Favourites({ favourites, deleteFavourite }) {
  function selectGenre(kind) {
    if (kind == null || kind === '') {
      return '';
    }

    // <!---- FAVOURITES FOREACH ARRAY---->
    const arraySplit = kind.split('-');
    let result = '';
    arraySplit.forEach((item, index) => {
      result += item[0].toUpperCase() + item.substring(1);
      if (index !== arraySplit.length - 1) {
        result += ' ';
      }
    });
    return result;
  }

  return (
    <div>
      <Table responsive hover variant='dark' size='md'>
        <thead className='TableHead'>
          <tr>
            <th colSpan='2'>Title</th>
            <th>Artist</th>
            <th colSpan='2'>Genre</th>
          </tr>
        </thead>
        <tbody>
          {favourites.map((favourite, index) => (
            <tr key={index}>
              <td colSpan='2'>{favourite.title}</td>
              <td>{favourite.artist}</td>
              <td>{selectGenre(favourite.kind)}</td>
              <td>
                <FontAwesomeIcon
                  icon={['fas', 'dumpster']}
                  className='fas fa-dumpster dumpsterBtn'
                  onClick={() => deleteFavourite(favourite.trackId)}
                />
              </td>
            </tr>
          ))}
        </tbody>
      </Table>
    </div>
  );
}

export default Favourites;
