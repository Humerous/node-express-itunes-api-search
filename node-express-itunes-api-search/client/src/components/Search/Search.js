import React from 'react';

// <!---- STYLE SHEET ---->
import '../Search/Search.css';

// <!---- IMPORT REACT STRAP ---->
import { Button, Input, Nav, Form } from 'reactstrap';

// <!---- SEARCH FAVOURITES FUNCTION---->
function Search(props) {
  return (
    <Nav className='SearchForm'>
      <Form onSubmit={props.searchAPIitunes}>
        <Input
          type='text'
          name='term'
          placeholder='Search titles or genres...'
        ></Input>

        <select name='media'>
          <option value='all'>All</option>
          <option value='music'>Music</option>
          <option value='audiobook'>Audio Book</option>
          <option value='musicVideo'>Music Video</option>
          <option value='movie'>Movie</option>
          <option value='tvShow'>TV Show</option>
          <option value='podcast'>Podcast</option>
          <option value='ebook'>E Book</option>
        </select>
        <Button color='primary'>Search</Button>
      </Form>
    </Nav>
  );
}

export default Search;
