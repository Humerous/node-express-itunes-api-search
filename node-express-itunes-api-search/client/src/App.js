import React from 'react';
import axios from 'axios';

// <!---- STYLE SHEET ---->
import './App.css';

// <!---- IMPORT REACT STRAP ---->
import { Container, Col, Row } from 'reactstrap';

// <!---- IMPORT COMPONENTS ---->
import Search from './components/Search/Search';
import SearchResults from './components/SearchResults/SearchResults';
import Favourites from './components/Favourites/Favourites';

// <!---- FONT AWESOME SCRIPT ---->
import { library } from '@fortawesome/fontawesome-svg-core';
import { faThumbsUp, faDumpster } from '@fortawesome/free-solid-svg-icons';
library.add(faThumbsUp, faDumpster);

class App extends React.Component {
  state = {
    results: [],
    favourites: [],
    error: undefined,
  };

  // ONLOAD FUNCTION
  componentWillMount = () => {
    axios.get('/api/favourites').then(
      (response) => {
        console.log(response.data);
        this.setState({
          favourites: response.data,
        });
      },
      (error) => {
        console.log(error);
        this.setState({
          error,
          favourites: [],
        });
      }
    );
  };

  //POST NEW FAVOURITE
  addFavourites = (result) => {
    axios
      .post('/api/favourites', {
        trackId: result.trackId,
        title: result.trackName,
        artist: result.artistName,
        kind: result.kind,
      })
      .then((response) => {
        const newFavourites = [...this.state.favourites];
        newFavourites.push(response.data);
        this.setState({ favourites: newFavourites });
      })
      .catch((error) => console.log(`${error}`));
  };

  //DELETE FAVOURITES
  deleteFavourite = (trackId) => {
    axios
      .delete('/api/favourites', { params: { 'track-id': trackId } })
      .then((res) => {
        const filteredFavourites = this.state.favourites.filter(
          (item) => item.trackId !== trackId
        );
        this.setState({ favourites: filteredFavourites });
      })
      .catch((error) => console.log(`Error message: ${error}`));
  };

  ////SEARCH ITUNE FUNCTION
  searchAPIitunes = async (e) => {
    e.preventDefault();
    const term = e.target.term.value;
    const media = e.target.media.value;
    axios
      .get(`/api/favourites/${term}/${media}`)
      .then((response) => {
        this.setState({ results: response.data });
      })
      .catch((error) => console.log(`Error message: ${error}`));
  };

  render() {
    return (
      <Container fluid>
        <Row>
          <Col sm={8} md={8} lg={8}>
            <div className='SearchOuterShell'>
              <div className='SearchOuterContent'>
                <div className='SearchOuterHeader'>
                  <h1 className='SearchOuterHeading'>Search</h1>
                  <Search searchAPIitunes={this.searchAPIitunes} />
                </div>
                <SearchResults
                  results={this.state.results}
                  error={this.state.error}
                  addFavourites={this.addFavourites}
                  favourites={this.state.favourites}
                />
              </div>
            </div>
          </Col>

          <Col sm={4} md={4} lg={4}>
            <div className='FavOuterShell'>
              <div className='FavOuterContent'>
                <div className='FavOuterHeader'>
                  <h1 className='FavHeading'>Favourites</h1>
                </div>
                <Favourites
                  favourites={this.state.favourites}
                  deleteFavourite={this.deleteFavourite}
                />
              </div>
            </div>
          </Col>
        </Row>
      </Container>
    );
  }
}

export default App;
