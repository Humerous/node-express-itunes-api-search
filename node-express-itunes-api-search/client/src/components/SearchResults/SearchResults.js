import React from 'react';

// <!---- STYLE SHEET ---->
import '../SearchResults/SearchResults.css';

//REACT-STRAP
import { Container, Row, Col } from 'reactstrap';

//FONT AWESOME
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

// <!---- SEARCH RESULTS FUNCTION---->
function SearchResults({ results, err, addFavourites, favourites }) {
  function selectGenre(kind) {
    if (kind == null || kind === '') {
      return '';
    }
    // <!---- SEARCH RESULTS FOREACH ARRAY---->
    const arraySplit = kind.split('-');
    let result = '';
    arraySplit.forEach((item, index) => {
      result += item[0].toUpperCase() + item.substring(1);
      if (index !== arraySplit.length - 1) {
        result += ' ';
      }
    });
    return result;
  }

  // <!---- SLECTED FAVOURITES FUNCTION ---->
  function isFavourite(trackId) {
    if (favourites.some((fav) => fav.trackId === trackId)) {
      return '#ec0101';
    } else {
      return 'white';
    }
  }

  return (
    <Container fluid>
      {err && <p className='ResultAllError'>{err}</p>}
      <Row xs={1} md={2} lg={3}>
        {results.map((result, index) => (
          <Col key={index} className='ResultOuterContainer'>
            <img
              className='ResultSmallImg'
              src={result.artworkUrl100}
              alt='search-result'
            />
            <div className='ResultAllDetails ml-2'>
              <p className='ResultsAllTitle'>{result.trackName}</p>
              <p className='ResultsAllImage'>{result.artistName}</p>
              <p className='ResultsAllMedia'>{selectGenre(result.kind)}</p>
            </div>
            <div className='ResultAllBtn mr-2'>
              <FontAwesomeIcon
                icon={['fas', 'thumbs-up']}
                className='fas fa-thumbs-up thumbBtn ml-2'
                onClick={() => addFavourites(result)}
                style={{ color: isFavourite(result.trackId) }}
              />
            </div>
          </Col>
        ))}
      </Row>
    </Container>
  );
}

export default SearchResults;
