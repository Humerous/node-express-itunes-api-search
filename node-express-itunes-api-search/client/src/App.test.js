import React from 'react';
import App from './App';
import { configure, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';

// <!---- IMPORT COMPONENTS ---->
import Search from './components/Search/Search';
import SearchResults from './components/SearchResults/SearchResults';
import Favourites from './components/Favourites/Favourites';

test('renders correctly', () => {
  const tree = renderer.create(<Search />).toJSON();

  expect(tree).toMatchSnapshot();
});
test('renders correctly', () => {
  const tree = renderer.create(<SearchResults />).toJSON();

  expect(tree).toMatchSnapshot();
});
test('renders correctly', () => {
  const tree = renderer.create(<Favourites />).toJSON();

  expect(tree).toMatchSnapshot();
});

configure({ adapter: new Adapter() });

describe('<App /> with no props', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render properly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
