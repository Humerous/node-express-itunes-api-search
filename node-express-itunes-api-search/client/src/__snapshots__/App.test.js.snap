// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> with no props should match the snapshot 1`] = `"<div class=\\"container-fluid\\"><div class=\\"row\\"><div class=\\"col-sm-8 col-md-8 col-lg-8\\"><div class=\\"SearchOuterShell\\"><div class=\\"SearchOuterContent\\"><div class=\\"SearchOuterHeader\\"><h1 class=\\"SearchOuterHeading\\">Search</h1><ul class=\\"SearchForm nav\\"><form class=\\"\\"><input type=\\"text\\" name=\\"term\\" placeholder=\\"Search titles or genres...\\" class=\\"form-control\\"/><select name=\\"media\\"><option value=\\"all\\">All</option><option value=\\"music\\">Music</option><option value=\\"audiobook\\">Audio Book</option><option value=\\"musicVideo\\">Music Video</option><option value=\\"movie\\">Movie</option><option value=\\"tvShow\\">TV Show</option><option value=\\"podcast\\">Podcast</option><option value=\\"ebook\\">E Book</option></select><button class=\\"btn btn-primary\\">Search</button></form></ul></div><div class=\\"container-fluid\\"><div class=\\"row row-cols-1 row-cols-md-2 row-cols-lg-3\\"></div></div></div></div></div><div class=\\"col-sm-4 col-md-4 col-lg-4\\"><div class=\\"FavOuterShell\\"><div class=\\"FavOuterContent\\"><div class=\\"FavOuterHeader\\"><h1 class=\\"FavHeading\\">Favourites</h1></div><div><div class=\\"table-responsive\\"><table variant=\\"dark\\" class=\\"table table-md table-hover\\"><thead class=\\"TableHead\\"><tr><th colSpan=\\"2\\">Title</th><th>Artist</th><th colSpan=\\"2\\">Genre</th></tr></thead><tbody></tbody></table></div></div></div></div></div></div></div>"`;

exports[`<App /> with no props should render properly 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row"
  >
    <div
      className="col-sm-8 col-md-8 col-lg-8"
    >
      <div
        className="SearchOuterShell"
      >
        <div
          className="SearchOuterContent"
        >
          <div
            className="SearchOuterHeader"
          >
            <h1
              className="SearchOuterHeading"
            >
              Search
            </h1>
            <ul
              className="SearchForm nav"
            >
              <form
                className=""
                onSubmit={[Function]}
              >
                <input
                  className="form-control"
                  name="term"
                  placeholder="Search titles or genres..."
                  type="text"
                />
                <select
                  name="media"
                >
                  <option
                    value="all"
                  >
                    All
                  </option>
                  <option
                    value="music"
                  >
                    Music
                  </option>
                  <option
                    value="audiobook"
                  >
                    Audio Book
                  </option>
                  <option
                    value="musicVideo"
                  >
                    Music Video
                  </option>
                  <option
                    value="movie"
                  >
                    Movie
                  </option>
                  <option
                    value="tvShow"
                  >
                    TV Show
                  </option>
                  <option
                    value="podcast"
                  >
                    Podcast
                  </option>
                  <option
                    value="ebook"
                  >
                    E Book
                  </option>
                </select>
                <button
                  aria-label={null}
                  className="btn btn-primary"
                  onClick={[Function]}
                >
                  Search
                </button>
              </form>
            </ul>
          </div>
          <div
            className="container-fluid"
          >
            <div
              className="row row-cols-1 row-cols-md-2 row-cols-lg-3"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-sm-4 col-md-4 col-lg-4"
    >
      <div
        className="FavOuterShell"
      >
        <div
          className="FavOuterContent"
        >
          <div
            className="FavOuterHeader"
          >
            <h1
              className="FavHeading"
            >
              Favourites
            </h1>
          </div>
          <div>
            <div
              className="table-responsive"
            >
              <table
                className="table table-md table-hover"
                variant="dark"
              >
                <thead
                  className="TableHead"
                >
                  <tr>
                    <th
                      colSpan="2"
                    >
                      Title
                    </th>
                    <th>
                      Artist
                    </th>
                    <th
                      colSpan="2"
                    >
                      Genre
                    </th>
                  </tr>
                </thead>
                <tbody />
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders correctly 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row"
  >
    <div
      className="col-sm-8 col-md-8 col-lg-8"
    >
      <div
        className="SearchOuterShell"
      >
        <div
          className="SearchOuterContent"
        >
          <div
            className="SearchOuterHeader"
          >
            <h1
              className="SearchOuterHeading"
            >
              Search
            </h1>
            <ul
              className="SearchForm nav"
            >
              <form
                className=""
                onSubmit={[Function]}
              >
                <input
                  className="form-control"
                  name="term"
                  placeholder="Search titles or genres..."
                  type="text"
                />
                <select
                  name="media"
                >
                  <option
                    value="all"
                  >
                    All
                  </option>
                  <option
                    value="music"
                  >
                    Music
                  </option>
                  <option
                    value="audiobook"
                  >
                    Audio Book
                  </option>
                  <option
                    value="musicVideo"
                  >
                    Music Video
                  </option>
                  <option
                    value="movie"
                  >
                    Movie
                  </option>
                  <option
                    value="tvShow"
                  >
                    TV Show
                  </option>
                  <option
                    value="podcast"
                  >
                    Podcast
                  </option>
                  <option
                    value="ebook"
                  >
                    E Book
                  </option>
                </select>
                <button
                  aria-label={null}
                  className="btn btn-primary"
                  onClick={[Function]}
                >
                  Search
                </button>
              </form>
            </ul>
          </div>
          <div
            className="container-fluid"
          >
            <div
              className="row row-cols-1 row-cols-md-2 row-cols-lg-3"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-sm-4 col-md-4 col-lg-4"
    >
      <div
        className="FavOuterShell"
      >
        <div
          className="FavOuterContent"
        >
          <div
            className="FavOuterHeader"
          >
            <h1
              className="FavHeading"
            >
              Favourites
            </h1>
          </div>
          <div>
            <div
              className="table-responsive"
            >
              <table
                className="table table-md table-hover"
                variant="dark"
              >
                <thead
                  className="TableHead"
                >
                  <tr>
                    <th
                      colSpan="2"
                    >
                      Title
                    </th>
                    <th>
                      Artist
                    </th>
                    <th
                      colSpan="2"
                    >
                      Genre
                    </th>
                  </tr>
                </thead>
                <tbody />
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
